### PR TITLE
client: abort the client if we couldn't invalidate dentry caches

### PR DESCRIPTION
**Pull request by lxbsz**
The option 'client_die_on_failed_dentry_invalidate' requires to kill
the client when fails to invalidate the dentry caches from kernel.
The CephFS client requires a mechanism to invalidate dentries in the
caller (e.g. the kernel for ceph-fuse) when capabilities must be recalled.
If the client cannot do this then the MDS cache cannot shrink which
can cause the MDS to fail.

Introduced by commit 8c778e79840f1aa9b9731e2ef20881da0d122fda.

Fixes: https://tracker.ceph.com/issues/57126
Signed-off-by: Xiubo Li <xiubli@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/client/Client.cc [hunk 1]
```diff
@@ -4583,7 +4583,11 @@ class C_Client_Remount : public Context  {
   explicit C_Client_Remount(Client *c) : client(c) {}
   void finish(int r) override {
     ceph_assert(r == 0);
-    client->_do_remount(true);
+    auto result = client->_do_remount(true);
+    bool abort_on_failure = result.second;
+    if (abort_on_failure) {
```
Comment (vshankar) [inline src/client/Client.cc:4588]: nit: just do

```
if (result.second) {
    ....
    ....
}
````

**Review by vshankar — COMMENTED** [review]
Otherwise looks good!

File: src/client/Client.cc (hunk 1 above)
Comment (lxbsz) [reply src/client/Client.cc:4588]: Looks good. Will fix it. Thanks!

**Review by vshankar — APPROVED** [review]

**Review by batrick — APPROVED** [review]

**Review by rishabh-d-dave — APPROVED** [review]
QA was successful - https://tracker.ceph.com/projects/cephfs/wiki/Main#2022-Aug-26. Waiting for CI jobs to pass.

**Review by rishabh-d-dave — APPROVED** [review]
QA was successful - https://tracker.ceph.com/projects/cephfs/wiki/Main#2022-Aug-26. Waiting for CI jobs to pass.

**Comment by lxbsz** [comment]
jenkins retest this please

**Review by kotreshhr — APPROVED** [review]

**Comment by rishabh-d-dave** [comment]
jenkins test make check

**Comment by rishabh-d-dave** [comment]
jenkins test make check arm64

**Comment by rishabh-d-dave** [comment]
jenkins test windows

**Comment by rishabh-d-dave** [comment]
https://jenkins.ceph.com/job/ceph-pull-requests/102910/consoleFull#408785433e840cee4-f4a4-4183-81dd-42855615f2c1

**Comment by rishabh-d-dave** [comment]
jenkins test make check

**Comment by rishabh-d-dave** [comment]
https://jenkins.ceph.com/job/ceph-pull-requests/102986/consoleFull#450684626c19247c4-fcb7-4c61-9a5d-7e2b9731c678

**Comment by rishabh-d-dave** [comment]
jenkins test make check

**Comment by rishabh-d-dave** [comment]
https://github.com/ceph/ceph/pull/47593#pullrequestreview-1086919928

**Pull request merged**